storage/series/buffer.go

```diff
@@ -337,6 +321,22 @@ func (b *dbBufferBucket) resetTo(start time.Time) {
 	b.drained = false
 }
 
+func (b *dbBufferBucket) canRead() bool {
+	return !b.drained && !b.empty
+}
+
+func (b *dbBufferBucket) needsReset(targetStart time.Time) bool {
+	return !b.start.Equal(targetStart)
+}
+
+func (b *dbBufferBucket) needsDrain(now time.Time, targetStart time.Time) bool {
+	retentionOpts := b.opts.RetentionOptions()
+	blockSize := retentionOpts.BlockSize()
+	bufferPast := retentionOpts.BufferPast()
```

Comment:
Hmm this might slow down the tick a little doing a little more indirection in this hot path.  Can we keep a reference to the `dbBuffer` then directly reference `b.buffer.blockSize` and `b.buffer.bufferPast`?